src/app/CMakeLists.txt

```diff
@@ -15,148 +43,136 @@ add_executable(qBittorrent
-        PROPERTIES
-            OUTPUT_NAME qbittorrent-nox
-    )
+    set_target_properties(qbt_app PROPERTIES OUTPUT_NAME qbittorrent-nox)
 endif()
 
 if (WEBUI)
-    target_link_libraries(qBittorrent PRIVATE qbt_webui)
-endif()
-
-# we have to include resources into the bundle
-if (APPLE)
-    set(OSX_RES_SRC_DIR "${qBittorrent_SOURCE_DIR}/dist/mac")
-    list(APPEND QBT_APP_RESOURCE_SOURCE
-        "${OSX_RES_SRC_DIR}/qt.conf"
-        "${OSX_RES_SRC_DIR}/qBitTorrentDocument.icns"
-        "${OSX_RES_SRC_DIR}/qbittorrent_mac.icns")
-    set_source_files_properties(
-        "${OSX_RES_SRC_DIR}/qt.conf"
-        "${OSX_RES_SRC_DIR}/qBitTorrentDocument.icns"
-        "${OSX_RES_SRC_DIR}/qbittorrent_mac.icns"
-        PROPERTIES
-            MACOSX_PACKAGE_LOCATION Resources)
-    set(QT_TR_DIR "${qBittorrent_SOURCE_DIR}/dist/qt-translations")
-
-    FILE(GLOB QT_TRANSLATIONS "${QT_TR_DIR}/qtbase_*.qm")
-    list(APPEND QT_TRANSLATIONS
-        ${QT_TR_DIR}/qt_fa.qm
-        ${QT_TR_DIR}/qt_gl.qm
-        ${QT_TR_DIR}/qt_lt.qm
-        ${QT_TR_DIR}/qt_pt.qm
-        ${QT_TR_DIR}/qt_sl.qm
-        ${QT_TR_DIR}/qt_sv.qm
-        ${QT_TR_DIR}/qt_zh_CN.qm
+    target_sources(qbt_app PRIVATE
+        ${QBT_WEBUI_QM_FILES}
+        ${qBittorrent_BINARY_DIR}/src/webui/www/translations/webui_translations.qrc # yes, it's supposed to be "*_BINARY_DIR"
     )
-    list(APPEND QBT_APP_RESOURCE_SOURCE ${QT_TRANSLATIONS})
-    set_source_files_properties(${QT_TRANSLATIONS} PROPERTIES MACOSX_PACKAGE_LOCATION translations)
+    target_link_libraries(qbt_app PRIVATE qbt_webui)
 endif()
 
-target_sources(qBittorrent PRIVATE ${QBT_QM_FILES} ${QBT_APP_RESOURCE_SOURCE})
-
-get_target_property(QBT_EXECUTABLE_NAME qBittorrent OUTPUT_NAME)
-
-if (APPLE)
-    set(qbt_BUNDLE_NAME ${QBT_EXECUTABLE_NAME})
-
-    # substitute @EXECUTABLE@ in dist/mac/Info.plist
-    set(EXECUTABLE ${qbt_BUNDLE_NAME})
-    configure_file(${qBittorrent_SOURCE_DIR}/dist/mac/Info.plist ${qBittorrent_BINARY_DIR}/dist/mac/Info.plist @ONLY)
-
-    set_target_properties(qBittorrent PROPERTIES
-        MACOSX_BUNDLE_BUNDLE_NAME "${qbt_BUNDLE_NAME}"
-        MACOSX_BUNDLE_INFO_PLIST ${qBittorrent_BINARY_DIR}/dist/mac/Info.plist
-    )
+if (GUI)
+    if ((CMAKE_SYSTEM_NAME STREQUAL "Windows") OR (CMAKE_SYSTEM_NAME STREQUAL "Darwin"))
+        qt5_import_plugins(qbt_app
+            INCLUDE Qt5::QSvgIconPlugin
+            INCLUDE Qt5::QSvgPlugin
+        )
+    endif()
 endif()
 
-# installation
-install(TARGETS qBittorrent
+# Installation
+# -----------------------------------------------------------------------------
+# -----------------------------------------------------------------------------
+install(TARGETS qbt_app
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     BUNDLE  DESTINATION .
     COMPONENT runtime
 )
 
-if (GUI AND APPLE)
-    find_package(Qt5Svg REQUIRED)
-    include(bundle)
+if (MSVC)
+    # GENEXP: honestly don't know how to do this without genexp
```

Comment:
Please remove this kind of comments. Those are only suitable on github reviews.

Reply:
fixed.